src/Npgsql/Internal/TypeHandlers/ArrayHandler.cs

```diff
@@ -496,15 +494,28 @@ public ArrayHandlerWithPsv(PostgresType arrayPostgresType, NpgsqlTypeHandler ele
 
         protected internal override async ValueTask<TRequestedArray> ReadCustom<TRequestedArray>(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription = null)
         {
-            if (ArrayTypeInfo<TRequestedArray>.ElementType == typeof(TElementPsv))
+            if (ArrayTypeInfo<TRequestedArray>.IsArray)
```

Comment:
I wonder if one day the JIT will do something nice with Type.IsArray, and then we should no longer cache this (and half this method would get elided). Doesn't seem to be the case today.

Reply:
For value types this is elided as it's a static readonly, but as you know for System.__Canon which array belongs to there is only one method body :/